src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionTemplate.java

```diff
@@ -127,9 +127,12 @@
         TreeFileArtifact outputTreeFileArtifact =
             ActionInputHelper.treeFileArtifactWithNoGeneratingActionSet(
                 outputTreeArtifact, PathFragment.create(outputName), artifactOwner);
-        TreeFileArtifact dotdFileArtifact =
-            ActionInputHelper.treeFileArtifactWithNoGeneratingActionSet(
-                dotdTreeArtifact, PathFragment.create(outputName + ".d"), artifactOwner);
+        TreeFileArtifact dotdFileArtifact = null;
+        if (dotdTreeArtifact != null) {
```

Comment:
Can you please also mark dotdTreeArtifact as @Nullable?

Reply:
Thanks, done!